### PR TITLE
Dev 1098

### PR DESCRIPTION
**Pull request by ahsinsiddique**

### CODE REVIEW

File: src/app/mail/mail-settings/mail-settings.component.html [hunk 1]
```diff
@@ -330,15 +330,23 @@ <h5 class="ui-header-subtitle text-dark mb-0"><strong translate="settings.genera
...
-                      <div ngbDropdown class="dropdown-sm">
-                        <button ngbDropdownToggle class="dropdown-toggle" id="timeZoneDropdownMenuButton"
-                                data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
-                          {{settings.timezone}}
-                        </button>
-                        <div ngbDropdownMenu aria-labelledby="timeZoneDropdownMenuButton" class="generic-dropdown-list">
-                          <a class="dropdown-item" *ngFor="let timezone of timezones"
-                             (click)="updateSettings('timezone',timezone.value)">{{timezone.value}}</a>
-                        </div>
+                      <div ngbDropdown class="dropdown-sm timezone-block">
```
Comment (atifsaddique211f) [inline src/app/mail/mail-settings/mail-settings.component.html]: remove ngbDropdown

File: src/app/mail/mail-settings/mail-settings.component.scss [hunk 2]
```diff
@@ -561,3 +561,27 @@ $error-red: #e94e4c;
     padding: 14px 1.5rem;
   }
 }
+
+.timezone-filter {
+  width: 100%;
+}
+
+.timezone-block {
```
Comment (atifsaddique211f) [inline src/app/mail/mail-settings/mail-settings.component.scss]: move it to component.styles.scc,
don't use ng deep
```
app-mail-settings {
.timezone-block {
....css here
}
}

File: src/app/mail/mail-settings/mail-settings.component.ts [hunk 3]
```diff
@@ -119,6 +128,18 @@ export class MailSettingsComponent implements OnInit, AfterViewInit, OnDestroy {
...
+
+    this.timeZoneFilteredOptions = this.timeZoneFilter.valueChanges
+      .pipe(
+        startWith(''),
+        map(name => name ? this._filterTimeZone(name) : this.timezones.slice())
+      );
+  }
+
+  private _filterTimeZone(name) {
+    const filterValue = name.toLowerCase();
```
Comment (atifsaddique211f) [inline src/app/mail/mail-settings/mail-settings.component.ts]: no need to declare `const filterValue` as it is used only on one place, use directly `name.toLowerCase(); `

File: src/app/mail/mail-settings/mail-settings.component.html (hunk 1 above)
Comment (ahsinsiddique) [reply src/app/mail/mail-settings/mail-settings.component.html]: removed.

File: src/app/mail/mail-settings/mail-settings.component.scss (hunk 2 above)
Comment (ahsinsiddique) [reply src/app/mail/mail-settings/mail-settings.component.scss]: moved to components, global css.

**Pull request merged**